src/drivers/net/dwc_gmac/dwc_gmac.c

```diff
@@ -395,17 +401,17 @@ static uint32_t dwc_setup_rxdesc(struct dwc_priv *priv, int idx) {
 
 	assert(idx < RX_DESC_QUANTITY);
 
-	desc = &priv->rxdesc_ring_paddr[idx];
+	desc = &priv->rxdesc_ring[idx];
 
 	memset(desc, 0, sizeof(*desc));
 	data_mem_barrier();
 
-	desc->basic.des2 = (uint32_t)&rx_buffers[idx][0];
+	desc->basic.des2 = (uint32_t)dma_map_single(NULL, &rx_buffers[idx][0], ETH_FRAME_LEN, DMA_FROM_DEVICE);
```

Comment:
Should we do symmetric `dma_unmap_single` call before mapping the same address again? Same for tx descriptors.